### ISSUE TITLE
Issue with History share via URL when using same history names

### ISSUE THREAD

**Issue by blankenberg** [galaxyproject/galaxy#14594]
Tested on Dev.

- Create a new history, name history "share testing"
- Upload file, rename dataset "share testing 0"
- Share history via link, slug is auto set to `.../h/share-testing`
- Verify that owner user can view via link (can access, is OK).
- Verify that non-logged in user can view via link (can access, is OK).
- Unshare history.
- Verify that owner user can view via link (can access, is maybe not correct).
- Verify that non-logged in user cannot view via link (cannot access, is OK. Gets unaccessible message, instead of 404 or 403, maybe is OK).
- Create a new history, name history "share testing"
- Upload file, rename dataset "share testing 1"
- Share history via link, slug is auto set to `.../h/share-testing`
- Verify that owner user can view via link (First history with "share testing 0" dataset is shown -- not OK).
- Verify that non-logged in user can view via link (cannot access, despite only the history with "share testing 1" being listed as shared/accessible and at the provided slugged link in the owner's account).

**Comment by hexylena** [comment]
have seen the same, previously. In effect slugs are "permanent" which isn't optimal. When unsharing, we should clear the slug maybe, thereby allowing its reuse later

**Comment by nsoranzo** [comment]
Maybe slugs should use the history uuid instead of its name?

**Comment by hexylena** [comment]
I'm not sure, they're intentionally "vanity" urls, they get used in papers/posters/etc and communicate useful information to the recipient

**Comment by nsoranzo** [comment]
Good point, but maybe we should leave that to URL shorteners?

**Comment by hexylena** [comment]
It's an option, but we'd be ripping out a well used, useful feature and telling folks they need to go somewhere else to do it which doesn't feel great, handing user tracking data over to a third party.

**Comment by blankenberg** [comment]
Agree, I think the vanity URLs are a really nice touch, and are currently used in several publications. 

I think probably the best course is to delete the slug when the history is unshared. Users can already manually change slugs whenever they like, so I don't think there is a concern about a change in mutability.

**Comment by hexylena** [comment]
Deleting slug on deletion would solve problems for me, I support it :)
